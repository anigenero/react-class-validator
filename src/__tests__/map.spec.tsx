import 'reflect-metadata';
import React from "react";
import {act, create} from "react-test-renderer";
import {ValidatorProvider} from "../index";
import {ContactForm} from "./ContactForm";

describe('context', () => {

    it('provider should mount correctly', () => {

        const tree = create(
            <ValidatorProvider options={{resultType: 'map'}}>
                <ContactForm/>
            </ValidatorProvider>
        ).toJSON();

        expect(tree).toMatchSnapshot();

    });

    it('validation success on form submit', async () => {

        const wrapper = create(
            <ValidatorProvider options={{resultType: 'map'}}>
                <ContactForm/>
            </ValidatorProvider>
        );

        const firstNameInput = wrapper.root.findByProps({id: 'fname-input'});
        await act(() =>
            firstNameInput.props.onChange({target: {value: 'Nick'}})
        );

        const lastNameInput = wrapper.root.findByProps({id: 'lname-input'});
        await act(() =>
            lastNameInput.props.onChange({target: {value: 'Fury'}})
        );

        const form = wrapper.root.findByType('form');
        await act(() =>
            form.props.onSubmit({
                preventDefault: jest.fn()
            })
        );

        expect(wrapper.toJSON()).toMatchSnapshot();

    });

    it('validation error on form submit', async () => {

        const wrapper = create(
            <ValidatorProvider options={{resultType: 'map'}}>
                <ContactForm/>
            </ValidatorProvider>
        );

        const firstNameInput = wrapper.root.findByProps({id: 'fname-input'});
        await act(() =>
            firstNameInput.props.onChange({target: {value: 'Nick'}})
        );

        const form = wrapper.root.findByType('form');
        await act(() =>
            form.props.onSubmit({
                preventDefault: jest.fn()
            })
        );

        expect(wrapper.toJSON()).toMatchSnapshot();

    });

    it('validation error on blur field', async () => {

        const wrapper = create(
            <ValidatorProvider options={{resultType: 'map'}}>
                <ContactForm/>
            </ValidatorProvider>
        );

        const firstNameInput = wrapper.root.findByProps({id: 'fname-input'});
        await act(() =>
            firstNameInput.props.onBlur()
        );

        expect(wrapper.toJSON()).toMatchSnapshot();

    });

    it('validation error custom handler', async () => {

        const wrapper = create(
            <ValidatorProvider options={{
                onErrorMessage() {
                    return ['this is a custom error']
                },
                resultType: 'map',
            }}>
                <ContactForm/>
            </ValidatorProvider>
        );

        const form = wrapper.root.findByType('form');
        await act(() =>
            form.props.onSubmit({
                preventDefault: jest.fn()
            })
        );

        expect(wrapper.toJSON()).toMatchSnapshot();

    });

});
