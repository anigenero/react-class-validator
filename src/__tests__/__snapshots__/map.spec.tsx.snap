// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`context provider should mount correctly 1`] = `
<form
  id="form"
  onSubmit={[Function]}
>
  <input
    id="fname-input"
    onBlur={[Function]}
    onChange={[Function]}
    value=""
  />
  <input
    id="lname-input"
    onBlur={[Function]}
    onChange={[Function]}
    value=""
  />
</form>
`;

exports[`context validation error custom handler 1`] = `
<form
  id="form"
  onSubmit={[Function]}
>
  <input
    id="fname-input"
    onBlur={[Function]}
    onChange={[Function]}
    value=""
  />
  <strong>
    this is a custom error
  </strong>
  <input
    id="lname-input"
    onBlur={[Function]}
    onChange={[Function]}
    value=""
  />
  <strong>
    this is a custom error
  </strong>
</form>
`;

exports[`context validation error on blur field 1`] = `
<form
  id="form"
  onSubmit={[Function]}
>
  <input
    id="fname-input"
    onBlur={[Function]}
    onChange={[Function]}
    value=""
  />
  <strong>
    First name cannot be empty
  </strong>
  <input
    id="lname-input"
    onBlur={[Function]}
    onChange={[Function]}
    value=""
  />
</form>
`;

exports[`context validation error on form submit 1`] = `
<form
  id="form"
  onSubmit={[Function]}
>
  <input
    id="fname-input"
    onBlur={[Function]}
    onChange={[Function]}
    value="Nick"
  />
  <input
    id="lname-input"
    onBlur={[Function]}
    onChange={[Function]}
    value=""
  />
  <strong>
    Last name cannot be empty
  </strong>
</form>
`;

exports[`context validation success on form submit 1`] = `
<form
  id="form"
  onSubmit={[Function]}
>
  <input
    id="fname-input"
    onBlur={[Function]}
    onChange={[Function]}
    value="Nick"
  />
  <input
    id="lname-input"
    onBlur={[Function]}
    onChange={[Function]}
    value="Fury"
  />
</form>
`;
